### ISSUE TITLE
Remote libvirt host. Waiting for domain to get an IP address

### ISSUE THREAD

**Issue by akam-it** [vagrant-libvirt/vagrant-libvirt#1160]
Hello,
I can't create VM on remote libvirt host. It stuck on "Waiting for domain to get an IP address"

### Expected behavior
VM should be created and provisioned

### Actual behavior
Creating VM stuck on getting IP address of VM however VM is successfully booted and IP address is exist in a virtual machine. (192.168.122.234/24) for example

### System configuration
**OS/Distro version:**: Centos 8 on the remote host and Fedora 32 on the local host (vagrant)

**Libvirt version:** remote libvirtd (libvirt) 6.0.0, local 6.1.0

**Output of `vagrant version; vagrant plugin list`:**
Installed Version: 2.2.9
vagrant-digitalocean (0.9.0, system)
vagrant-libvirt (0.0.45, system)

**Output of `VAGRANT_LOG=debug vagrant ... --provider=libvirt`**
```
==> kamailio1: Creating shared folders metadata...
 INFO warden: Calling IN action: #<VagrantPlugins::ProviderLibvirt::Action::CreateNetworks:0x00007f145c286000>
 INFO warden: Calling IN action: #<VagrantPlugins::ProviderLibvirt::Action::CreateNetworkInterfaces:0x00007f145c248228>
 INFO create_network_interfaces: Using vagrant-libvirt at 192.168.121.0/24 as the management network nat is the mode
DEBUG create_network_interfaces: In config found network type forwarded_port options {:guest=>22, :host=>2222, :host_ip=>"127.0.0.1", :id=>"ssh", :auto_correct=>true, :protocol=>"tcp"}
DEBUG create_network_interfaces: Adapter not specified so found slot 0
 INFO create_network_interfaces: Setting up public interface using device virbr0 in mode bridge
 INFO create_network_interfaces: Creating network interface eth0 connected to network public.
 INFO warden: Calling IN action: #<VagrantPlugins::ProviderLibvirt::Action::SetBootOrder:0x00007f145c1e2338>
 INFO warden: Calling IN action: #<VagrantPlugins::ProviderLibvirt::Action::StartDomain:0x00007f145c180480>
 INFO interface: info: Starting domain.
 INFO interface: info: ==> kamailio1: Starting domain.
==> kamailio1: Starting domain.
 INFO warden: Calling IN action: #<VagrantPlugins::ProviderLibvirt::Action::WaitTillUp:0x0000557f44930210>
DEBUG wait_till_up: Searching for IP for MAC address: 52:54:00:70:63:81
 INFO interface: info: Waiting for domain to get an IP address...
 INFO interface: info: ==> kamailio1: Waiting for domain to get an IP address...
==> kamailio1: Waiting for domain to get an IP address...
 INFO retryable: Retryable exception raised: #<Fog::Errors::TimeoutError: The specified wait_for timeout (2 seconds) was exceeded>
 INFO retryable: Retryable exception raised: #<Fog::Errors::TimeoutError: The specified wait_for timeout (2 seconds) was exceeded>

```

**A Vagrantfile to reproduce the issue:**
```
Vagrant.configure("2") do |config|
  config.vm.provider :libvirt do |libvirt|
    libvirt.host = "192.168.88.10"
    libvirt.connect_via_ssh = true
    libvirt.socket = "/var/run/libvirt/libvirt-sock"
    #libvirt.management_network_device = 'bridge0'
  end

  # kamailio1
  config.vm.define "kamailio1" do |v|
    v.vm.box = "centos/7"
    v.vm.hostname = "kamailio1"
#    v.vm.network "public_network", :dev=>"bridge0"

    v.vm.provider "libvirt" do |vb|
      vb.memory = "512"
    end

    v.vm.provision "ansible" do |ansible|
      ansible.playbook = ".ansible/kamailio.yml"
    end
  end

end

```

**Are you using upstream vagrant package or your distros package?**
Distro

**Comment by abbbi** [comment]
There are two ways the plugin trys to find the ip address

1) via /proc/net/arp table, which obviously wont work here as you are spinning up the system on a remote system2
2) via the net-dhcp-leases function by libvirt.

is the mac address and ip of the system that was spinned up on the remote system listed in the output of the
vagrant-libvirt dhcp leases?

virsh net-dhcp-leases vagrant-libvirt

**Comment by akam-it** [comment]
Hello, thanks for reply.

I have changed Vagrantfile, added
libvirt.management_network_name = "default"
management_network_address = "192.168.122/24"

I have executed "virsh net-dhcp-leases vagrant-libvirt" on the remote machine and I can see mac and IP..
![image](https://user-images.githubusercontent.com/1487709/97421106-2cb90100-191d-11eb-8bf7-7751c8045987.png)

but vagrant doesn't
![image](https://user-images.githubusercontent.com/1487709/97421330-6f7ad900-191d-11eb-8b68-689d5d8c64fa.png)

**Comment by akam-it** [comment]
anyway my desktop machine will not ping this 192.168.122.0/24 network
should I use bridged network for management?

**Comment by abbbi** [comment]
The network on the remote machine is NATTET, that means by default the VM ip is not reachable from an external system, because the libvirt network it is connected to is not routed accross hosts. 

You need to setup a routed network here to make the management network ip beeing reachable from external ressources.

**Comment by electrofelix** [comment]
@akam-it there should be no need to use a bridged network, the ssh-info returned to vagrant by vagrant-libvirt should contain a `ProxyCommand` entry with the required connection details to have ssh connect to the remote machine before trying to connect to the VM on the remote host. This allows connecting as though the machine is directly accessible even if it's networking is only accessible to the remote host.

The version of the plugin you're reporting is a bit out of date (0.0.45 was released Oct 2018), there was definitely some work done around the remote options to correct buggy behaviour, so I'd recommend updating to the most recent release (0.3.0).

There does appear to be a bug within vagrant and it's use of SSH channels that only appears to get tickled when the proxycommand is set at the moment so you'll need to set `config.ssh.insert_key = false` to workaround for now.

**Comment by electrofelix** [comment]
Actually appears the problem is the use of `nc %h %p` as part of the ProxyCommand setting, switching to `-W %h:%p` works just fine. So looks like we'll need to release a fix as well, though you can use the `config.ssh.insert_key = false` as a temporary workaround once on a new enough release.

**Comment by electrofelix** [comment]
Turns out this is a duplicate of #921, going to close this issue and focus on trying to implement a proper fix for #921 in the near future. Likely will need to allow a config option for for ProxyCommand for users to provide a custom one in cases where it's not working as expected.
